src/server/pfs/server/driver_new_storage.go

```diff
@@ -339,23 +339,28 @@ func deserializeShard(shardAny *types.Any) (*pfs.Shard, error) {
 	return shard, nil
 }
 
-// (bryce) it might potentially make sense to exit if an error occurs in this function
-// because each pachd instance that errors here will lose its compaction worker without an obvious
-// notification for the user (outside of the log message).
-// (bryce) ^ maybe just a retry would be good enough.
 func (d *driver) compactionWorker() {
 	w := work.NewWorker(d.etcdClient, d.prefix, storageTaskNamespace)
-	if err := w.Run(context.Background(), func(ctx context.Context, subtask *work.Task) error {
-		shard, err := deserializeShard(subtask.Data)
+	backoffStrat := backoff.NewExponentialBackOff()
+	backoffStrat.MaxElapsedTime = 0 // don't give up, compactionWorker should not exit
```

Comment:
Just a stylistic note, I tend to prefer not adding spaces between lines of code and instead use comments to separate logical groups (or to just add notes). Also, I don't really comment on the same line as code. In the interest of consistency, we should probably try and follow the same style.